build-tool.sh

```diff
@@ -0,0 +1,160 @@
+    local file=$1
+    if [[ $file == *.ts ]]; then
+        echo "Deno"
+    elif [[ $file == *.py ]]; then
+        echo "Python"
+    fi
+}
+
+# Process each tool directory
+echo "Processing tools..."
+for tool_dir in tools/*/; do
+    if [ -d "$tool_dir" ]; then
+        # Get the tool name from directory
+        tool_name=$(basename "$tool_dir")
+        
+        echo "Processing ${tool_name}"
+        
+        # Check for either tool.ts or tool.py
+        tool_file=""
+        if [ -f "${tool_dir}tool.ts" ]; then
+            tool_file="${tool_dir}tool.ts"
+        elif [ -f "${tool_dir}tool.py" ]; then
+            tool_file="${tool_dir}tool.py"
+        fi
+
+        # Check if required files exist
+        if [ -z "$tool_file" ] || [ ! -f "${tool_dir}metadata.json" ]; then
+            echo $tool_file $tool_dir
+            echo "Error: Missing required files in ${tool_dir}"
+            continue
+        fi
+
+        # Create packages directory if it doesn't exist
+        mkdir -p "packages/${tool_name}"
+
+        # Read metadata.json
+        metadata_content=$(cat "${tool_dir}metadata.json")
+        
+        # Read tool content
+        tool_content=$(cat "$tool_file")
+
+        # Get tool type based on file extension
+        tool_type=$(get_tool_type "$tool_file")
+
+        # Convert directory name to toolkit name
+        toolkit_name=$(convert_to_toolkit_name "$tool_name")
+
+        # Create tool.json content
+        # Set the code key based on tool type
+        code_key="js_code"
+        if [ "$tool_type" = "Python" ]; then
+            code_key="py_code"
+        fi
+
+        jq -n \
+            --arg type "$tool_type" \
+            --arg toolkit_name "${toolkit_name}" \
+            --arg tool_content "$tool_content" \
+            --arg code_key "$code_key" \
+            --argjson metadata "$metadata_content" \
+            '{
+            type: $type,
+            content: [
+                {
+                toolkit_name: $toolkit_name,
+                name: $metadata.name,
+                author: $metadata.author,
+                ($code_key): $tool_content,
+                tools: $metadata.tools,
+                config: $metadata.configurations,
+                description: $metadata.description,
+                keywords: $metadata.keywords,
+                input_args: $metadata.parameters,
+                output_arg: {json: ""},
+                activated: false,
+                embedding: null,
+                result: $metadata.result,
+                sql_tables: $metadata.sqlTables,
+                sql_queries: $metadata.sqlQueries,
+                file_inbox: null,
+                oauth: $metadata.oauth
+                },
+                false
+            ]
+            }' > "packages/${tool_name}/tool.json"
+
+        echo "Generated tool.json for ${tool_name}"
+
+        # Create zip file
+        cd "packages/${tool_name}"
+        zip "${toolkit_name}.zip" tool.json
```

Comment:
creating the .zip here might be hard to maintain in the long term if we would have to keep in sync the node and this tool 👁️ the export endpoint can be called after v2/set_playground_tool

```
## Export Tool
curl "http://localhost:9950/v2/export_tool?tool_key_path=local%3A%3A%3Acheck_files2_shinkai%3A%3A%3Acheck_files2" \
     -H 'Authorization: Bearer debug' \
     -H 'Content-Type: application/json; charset=utf-8' > file.zip
```

Reply:
Created the .zip file directly from node with this approach. First I import it to the node and then I download it as a .zip file